File: exclusapdedadosPHP/livroseautores.php
<?php // nome: maxwell
$servername = "localhost";
$username = "USER";
$password = "PASS";
$dbname = "DATABASE";

$conn = new mysqli($servername, $username, $password, $dbname);

if ($conn->connect_error) {
    die("Erro de conexão: " . $conn->connect_error);
}

$id_livro = 1; // Insira o ID do livro que deseja excluir aqui

try {
    $conn->begin_transaction();

    // Exclui as associações do livro com autores na tabela `autores`
    $sql = "DELETE FROM autores WHERE id_livro = ?";
    $stmt = $conn->prepare($sql);
    $stmt->bind_param("i", $id_livro);
    $stmt->execute();

    // Exclui o livro específico da tabela `livros`
    $sql = "DELETE FROM livros WHERE id = ?";
    $stmt = $conn->prepare($sql);
    $stmt->bind_param("i", $id_livro);
    $stmt->execute();

    $conn->commit();

    echo "Livro e associações com autores excluídos com sucesso!";

} catch (Exception $e) {
    echo "Erro: " . $e->getMessage();

} finally {
    $stmt->close();
    $conn->close();
}
?>

File: exclusaodedadosJAVA/livroseautores.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.SQLException;
//Nome:Joao Carlos

public class livroseautores {
    public static void main(String[] args) {
        final String url = "jdbc:mysql://localhost:3306/nomedobanco";
        final String user = "nome de usuario";
        final String password = "senha";

        try (Connection conn = DriverManager.getConnection(url, user, password)) {
            conn.setAutoCommit(false);

            int idLivro = 1; // Id do livro a ser excluído

            // Remove todas as associações do livro com autores
            String sql = "DELETE FROM autores_livros WHERE livro_id = ?";
            try (PreparedStatement pstmt = conn.prepareStatement(sql)) {
                pstmt.setInt(1, idLivro);
                pstmt.executeUpdate();
            }

            // Exclui o livro
            sql = "DELETE FROM livros WHERE id = ?";
            try (PreparedStatement pstmt = conn.prepareStatement(sql)) {
                pstmt.setInt(1, idLivro);
                pstmt.executeUpdate();
            }

            conn.commit();
            System.out.println("Livro e autores associados excluídos com sucesso!");

        } catch (SQLException e) {
            System.out.println("Erro ao excluir livro e autores associados: " + e.getMessage());
        }
    }
}

